File: MoneyFlowAPI/Application/DTOs/DTO_CategoriesPage.cs
namespace MoneyFlowAPI.Application.DTOs
{
    public class DTO_CategoriesPage
    {
        public List<DTO_Category> Categories { get; set; } = new();

        public int TotalCategoriesCount { get; set; }

        public int MonthlyCategoriesCount { get; set; }

        public string LatestCategoryName { get; set; } = "Sem dados";

        public DateTime? LatestCategoryCreatedAt { get; set; }
    }
}

File: MoneyFlowUI/src/app/shared/dtos/DTO_CategoriesPage.ts
import { DTO_Category } from './DTO_Category';

export class DTO_CategoriesPage {
    categories: DTO_Category[] = [];
    totalCategoriesCount = 0;
    monthlyCategoriesCount = 0;
    latestCategoryName = 'Sem dados';
    latestCategoryCreatedAt: string | null = null;
}
